Firestore/core/src/firebase/firestore/nanopb/byte_string.h

```diff
@@ -39,6 +44,11 @@ class ByteString : public util::Comparable<ByteString> {
  public:
```

Comment:
Can you please expand on the class comment? In particular, it might be worth that this class owns its memory,
has deep copy semantics, and under most circumstances creates a copy of the input given to its constructor. 

Additionally:
* from its interface, it seems like it's much more of a byte buffer than a string, because it doesn't provide any text manipulation. The only exception I noticed is that comparison is lexicographical;
* what's the intended use for this class? Why would somebody use it instead of manipulating Nanopb types directly? I'm mostly asking due to the presence of `release` function.